prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <limits.h>
+
+int generate_primes(int num) {
+    int cnt = 0;
+    printf("Prime Factors: ");
```

Comment:
The number of elements filled in that array will be equal to number of prime factors of the number. So yes, `cnt` will be restricted.

Hope this makes sense to you - 
```C
int generate_primes(int num, int *prime_factors) {
    int cnt = 0;
    ...
    // Calculate primes and store it in prime_factors, incrementing cnt everytime.
    ...
    return cnt;
}

...
int n = 10;
int prime_factors[n];  // n will be an upper bound for the number of prime factors for n (but maybe we can improve here)?
int num_primes = generate_primes(n, prime_factors);
...
```